web/src/components/network/WifiSelectorPage.jsx

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) [2022] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useCallback, useEffect, useState } from "react";
+import { useInstallerClient } from "~/context/installer";
+import { NetworkEventTypes } from "~/client/network";
+import { Page } from "~/components/core";
+import { WifiNetworksListPage } from "~/components/network";
+import { _ } from "~/i18n";
+import { DeviceState } from "~/client/network/model";
+import { Grid, GridItem, Timestamp } from "@patternfly/react-core";
+import { useLoaderData } from "react-router-dom";
+import { useLocalStorage } from "~/utils";
+
+const networksFromValues = (networks) => Object.values(networks).flat();
+const baseHiddenNetwork = { ssid: undefined, hidden: true };
+
+// FIXME: use a reducer
+
+function WifiSelectorPage() {
+  const { network: client } = useInstallerClient();
+  const { connections: initialConnections, devices: initialDevices, accessPoints, networks: initialNetworks } = useLoaderData();
+  const [data, saveData] = useLocalStorage("agama-network", { selectedWifi: null });
+  const [selected, setSelected] = useState(data.selectedWifi);
```

Comment:
JFTR, we agreed to re-evaluate how to keep the state in the future (especially given that we are considering a state management library).